### PR TITLE
DT-2556: log event to AppInsights; remove Hidden Swagger annotation

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/DT-2556

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/OffenderDatesService.java [hunk 1]
```diff
@@ -71,6 +74,9 @@ public SentenceCalcDates updateOffenderKeyDates(Long bookingId, RequestToUpdateO
                 .judiciallyImposedSentenceLength(keyDatesFromPayload.getSentenceLength())
                 .build();
         offenderBooking.addSentenceCalculation(sentenceCalculation);
+
+        telemetryClient.trackEvent("OffenderKeyDatesUpdated", ImmutableMap.of("bookingId", bookingId.toString(), "calculationUuid", requestToUpdateOffenderDates.getCalculationUuid().toString()), null);
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/service/OffenderDatesService.java:78]: Log the user

**Review by Mjwillis — CHANGES_REQUESTED** [review]
I would log the user too

**Review by Mjwillis — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/service/OffenderDatesService.java (hunk 1 above)
Comment (neilmendum) [reply src/main/java/uk/gov/justice/hmpps/prison/service/OffenderDatesService.java]: Good point, updated

**Pull request merged**